src/test/java/core/basesyntax/service/impl/DataProcessingServiceTest.java

```diff
@@ -0,0 +1,88 @@
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                new Fruit("banana"), 10));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                new Fruit("apple"), 20));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                new Fruit("cherry"), 100));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                new Fruit("apple"), 70));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                new Fruit("banana"), 80));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                new Fruit("banana"), 10));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.RETURN,
+                new Fruit("apple"), 30));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                new Fruit("apple"), 20));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                new Fruit("banana"), 20));
+        fruitTransaction.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                new Fruit("banana"), 40));
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<FruitTransaction.Operation, OperationHandler> operationServiceMap = new HashMap<>();
+        operationServiceMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        operationServiceMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandlerService());
+        operationServiceMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseHandler());
+        operationServiceMap.put(FruitTransaction.Operation.RETURN, new ReturnHandlerService());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationServiceMap);
+        dataProcessingService = new DataProcessingServiceImpl(operationStrategy);
+        ReportCreatorService reportCreatorService = new ReportCreatorServiceImpl();
+    }
+
+    @Test
+    public void processingDate_correctData_ok() {
+        Map<Fruit, Integer> expectedMap = new HashMap<>();
+        expectedMap.put(new Fruit("banana"), 100);
+        expectedMap.put(new Fruit("apple"), 100);
+        expectedMap.put(new Fruit("cherry"), 100);
+        Map<Fruit, Integer> actualMap = Storage.fruitBalance;
+        dataProcessingService.processingDate(fruitTransaction);
+        Assert.assertEquals(expectedMap.toString(), actualMap.toString());
+    }
+
+    @Test
+    public void processingDate_emptyData_notOk() {
+        List<FruitTransaction> emptyReport = new ArrayList<>();
+        try {
+            dataProcessingService.processingDate(emptyReport);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown is pats was wrong");
```

Comment:
Let's use already existing methods of junit library
```suggestion
            assertThrows(RuntimeException.class, 
                () -> dataProcessingService.processingDate(emptyReport), "RuntimeException should be thrown is pats was wrong")
```
And the message is unnecessary